### ISSUE TITLE
Mocking with jest doesn't work

### ISSUE THREAD

**Issue by bennyk8y** [swc-project/swc-node#713]
Hi, this is `jest.config.template.js`:
```js
module.exports = {
  testEnvironment: 'node',
  testMatch: ['**/*.test.ts', '**/**/**/*.test.ts', '**/**/**/*.spec.ts'],
  transform: {
    '^.+\\.ts?$': [
      '@swc-node/jest',
      {
        sourcemap: true,
        experimentalDecorators: true,
        emitDecoratorMetadata: true,
        keepClassNames: true,
      },
    ],
  },
};
```

this is the test file:
```
import { getBankTransactionOrFail } from '../../models/queries/getBankTransaction';

jest.mock('../../models/queries/getBankTransaction');

describe('fetchRelatedTransactionsForClassification', () => {

  it('no related entities', async () => {
    (getBankTransactionOrFail as jest.Mock).mockResolvedValueOnce(makeBankTransaction());
```

result an error of `TypeError: _getBankTransaction.getBankTransactionOrFail.mockResolvedValueOnce is not a function`.

**
"@swc-node/jest": "1.6.5",
"jest": "28.1.3"
**

**Comment by meskill** [comment]
The issue is related to https://github.com/swc-project/swc-node/issues/700

In previous major version @swc-node/jest@1.5.5 mocks are working right

I'm expecting that https://github.com/swc-project/swc-node/pull/714 will fix the problem

**Comment by meskill** [comment]
Please, check the latest version

**Comment by rosmcmahon** [comment]
no change for Sinon

**Comment by rosmcmahon** [comment]
yeah, catching up on this again. basically sinon recognising that swc is building the module exports in a more correct way, but that it's out of scope for their project to support this.

**Comment by fatso83** [comment]
The problem is that Jest is a combination of many things: 
- test runner (a la Mocha)
- stubs and spies (a la Sinon)
- module interception (a la Proxyquire)

By itself, nether Jest nor Sinon can know that an object is a module with exports and what you want to do with it 🤷 Both require the user to know what they are looking at.

**Comment by rosmcmahon** [comment]
thanks @fatso83, sorry for the delay.
yeah, i was adding swc to an older codebase of mine. the solution was to remove swc, and revert to `transpileOnly` with type-checking step. 
IMO if i'm using proxyquire then i'm doing something wrong as i should write my code with testing in mind. i even avoid stubs these days

**Comment by fatso83** [comment]
In [the article I have out for review](https://github.com/sinonjs/sinon/pull/2540) I discuss three different approaches to exactly this issue, one of which is the tool independent route of exposing ways of injecting dependencies. I guess you are doing something similar to what I propose there?

Would really like some extra set of eyes on this, so if you could spare five minutes I can perhaps help quite a few people in avoiding the trouble you experienced 🙏
